roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -50,17 +52,56 @@ pub struct Sniffer {
     upstream_address: SocketAddr,
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
+    intercept_messages: Vec<InterceptMessage>,
+}
+
+#[derive(Debug, Clone)]
+pub struct InterceptMessage {
+    direction: MessageDirection,
+    expected_message_type: MsgType,
+    response_message: PoolMessages<'static>,
+    response_message_type: MsgType,
+    break_on: bool,
```

Comment:
what is the purpose of `break_on`?

under which circumstances can it be useful?

Reply:
when `break_on` is set to `true`, the sniffer will exist after sending the mutated message, which will result in closing the connection between downstream and upstream. Can be useful if we dont want the roles exchange messages after we mutate a message

Reply:
I assume you mean:

> the sniffer will EXIT after sending the mutated message

right?

Reply:
I still don't see how this could be useful. I am yet to encounter a scenario where it would make sense to break the connection and still continue the test.

Could you give some concrete example?